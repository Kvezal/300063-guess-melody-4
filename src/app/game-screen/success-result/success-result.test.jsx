import React from "react";
import render from "react-test-renderer";
import Enzyme, {mount} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import SuccessResult from "./success-result";


Enzyme.configure({
  adapter: new Adapter(),
});


describe(`SuccessResultComponent`, () => {
  test(`should render success component`, () => {
    const tree = render
      .create(
          <SuccessResult
            rightAnswers={10}
            mistakes={0}
            onResetLinkClick={() => {}}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test(`should have a success title`, () => {
    const resultComponent = mount(
        <SuccessResult
          rightAnswers={0}
          mistakes={0}
          onResetLinkClick={() => {}}
        />
    );
    const failedResult = resultComponent.find(`p.result__total--fail`);
    const successResult = resultComponent.find(`p.result__total`);
    expect(failedResult).toHaveLength(0);
    expect(successResult).toHaveLength(1);
  });

  test(`should have right answers count`, () => {
    const rightAnswers = 10;
    const resultComponent = mount(
        <SuccessResult
          rightAnswers={rightAnswers}
          mistakes={0}
          onResetLinkClick={() => {}}
        />
    );
    const result = resultComponent.find(`p.result__total`);
    expect(result.text().includes(rightAnswers)).toBeTruthy();
  });

  test(`should have right mistakes count`, () => {
    const mistakes = 2;
    const resultComponent = mount(
        <SuccessResult
          rightAnswers={8}
          mistakes={mistakes}
          onResetLinkClick={() => {}}
        />
    );
    const result = resultComponent.find(`p.result__total`);
    expect(result.text().includes(mistakes)).toBeTruthy();
  });
});
