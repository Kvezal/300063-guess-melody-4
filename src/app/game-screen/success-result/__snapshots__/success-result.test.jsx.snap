// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessResultComponent should render success component 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    10
     вопросов и совершили 
    0
     ошибки
  </p>
  <button
    className="replay"
    type="button"
  >
    Сыграть ещё раз
  </button>
</section>
`;
